experimental/slo/metrics.go

```diff
@@ -0,0 +1,227 @@
+package slo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// Metrics is an abstraction for collecting metrics
+type Metrics struct {
+	DSName   string
+	DSType   string
+	Endpoint Endpoint
+}
+
+// Duration is stored in the Context and used to collect metrics
+type Duration struct {
+	Value      float64
+	Status     Status
+	Source     Source
+	StatusCode int
```

Comment:
Nit. Suggest these to be non-exported since otherwise someone could in theory read/write the fields without taking the mutex into account.